bundle/config/mutator/apply_transforms_test.go

```diff
@@ -0,0 +1,219 @@
+package mutator_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/config"
+	"github.com/databricks/cli/bundle/config/mutator"
+	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTransformPrefix(t *testing.T) {
+	tests := []struct {
+		name   string
+		prefix string
+		job    *resources.Job
+		want   string
+	}{
+		{
+			name:   "add prefix to job",
+			prefix: "prefix-",
+			job: &resources.Job{
+				JobSettings: &jobs.JobSettings{
+					Name: "job1",
+				},
+			},
+			want: "prefix-job1",
+		},
+		{
+			name:   "add empty prefix to job",
+			prefix: "",
+			job: &resources.Job{
+				JobSettings: &jobs.JobSettings{
+					Name: "job1",
+				},
+			},
+			want: "job1",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b := &bundle.Bundle{
+				Config: config.Root{
+					Resources: config.Resources{
+						Jobs: map[string]*resources.Job{
+							"job1": tt.job,
+						},
+					},
+					Transform: config.Transforms{
+						Prefix: tt.prefix,
+					},
+				},
+			}
+
+			mutator := mutator.ApplyTransforms()
+			diag := mutator.Apply(context.Background(), b)
+
+			if diag.HasError() {
+				t.Fatalf("unexpected error: %v", diag)
+			}
+
+			if got := b.Config.Resources.Jobs["job1"].Name; got != tt.want {
+				t.Errorf("got %v, want %v", got, tt.want)
+			}
```

Comment:
Sure, I can make these changes. The code for this test is 100% API generated and just reviewed by me since it's a bit redundant compared to the process_target_mode_test. The style here isn't 100% how I'd write it but seemed nice and readable to me for a test.